[2024-03-30 22:20:08,235] torch.distributed.run: [WARNING] 
[2024-03-30 22:20:08,235] torch.distributed.run: [WARNING] *****************************************
[2024-03-30 22:20:08,235] torch.distributed.run: [WARNING] Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
[2024-03-30 22:20:08,235] torch.distributed.run: [WARNING] *****************************************
wandb: Currently logged in as: 3jp4rk. Use `wandb login --relogin` to force relogin
wandb: Currently logged in as: 3jp4rk. Use `wandb login --relogin` to force relogin
wandb: Tracking run with wandb version 0.16.5
wandb: Run data is saved locally in /home/ubuntu/ejpark/Condenser/wandb/run-20240330_222012-vky4ar5t
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run woven-mountain-45
wandb: ‚≠êÔ∏è View project at https://wandb.ai/3jp4rk/Condenser
wandb: üöÄ View run at https://wandb.ai/3jp4rk/Condenser/runs/vky4ar5t/workspace
wandb: Tracking run with wandb version 0.16.5
wandb: Run data is saved locally in /home/ubuntu/ejpark/Condenser/wandb/run-20240330_222012-w6obiu37
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run effortless-dream-46
wandb: ‚≠êÔ∏è View project at https://wandb.ai/3jp4rk/Condenser
wandb: üöÄ View run at https://wandb.ai/3jp4rk/Condenser/runs/w6obiu37/workspace
wandb: WARNING Calling wandb.login() after wandb.init() has no effect.
wandb: WARNING Calling wandb.login() after wandb.init() has no effect.
03/30/2024 22:20:55 - WARNING - __main__ -   Process rank: 0, device: cuda:0, n_gpu: 1distributed training: True, 16-bits training: False
03/30/2024 22:20:55 - INFO - __main__ -   Training/evaluation parameters CoCondenserPreTrainingArguments(
_n_gpu=1,
accelerator_config={'split_batches': False, 'dispatch_batches': None, 'even_batches': True, 'use_seedable_sampler': True},
adafactor=False,
adam_beta1=0.9,
adam_beta2=0.999,
adam_epsilon=1e-08,
auto_find_batch_size=False,
bf16=False,
bf16_full_eval=False,
cache_chunk_size=-1,
data_seed=None,
dataloader_drop_last=False,
dataloader_num_workers=0,
dataloader_persistent_workers=False,
dataloader_pin_memory=True,
dataloader_prefetch_factor=None,
ddp_backend=None,
ddp_broadcast_buffers=None,
ddp_bucket_cap_mb=None,
ddp_find_unused_parameters=None,
ddp_timeout=1800,
debug=[],
deepspeed=None,
disable_tqdm=False,
dispatch_batches=None,
do_eval=False,
do_predict=False,
do_train=True,
eval_accumulation_steps=None,
eval_delay=0,
eval_steps=5000.0,
evaluation_strategy=no,
fp16=False,
fp16_backend=auto,
fp16_full_eval=False,
fp16_opt_level=O1,
fsdp=[],
fsdp_config={'min_num_params': 0, 'xla': False, 'xla_fsdp_v2': False, 'xla_fsdp_grad_ckpt': False},
fsdp_min_num_params=0,
fsdp_transformer_layer_cls_to_wrap=None,
full_determinism=False,
gradient_accumulation_steps=1,
gradient_checkpointing=False,
gradient_checkpointing_kwargs=None,
greater_is_better=None,
group_by_length=False,
half_precision_backend=auto,
hub_always_push=False,
hub_model_id=None,
hub_private_repo=False,
hub_strategy=every_save,
hub_token=<HUB_TOKEN>,
ignore_data_skip=False,
include_inputs_for_metrics=False,
include_num_input_tokens_seen=False,
include_tokens_per_second=False,
jit_mode_eval=False,
label_names=None,
label_smoothing_factor=0.0,
learning_rate=5e-05,
length_column_name=length,
load_best_model_at_end=False,
local_rank=0,
log_level=passive,
log_level_replica=warning,
log_on_each_node=True,
logging_dir=co_condenser_pretrain/runs/Mar30_22-20-20_gpu-1,
logging_first_step=False,
logging_nan_inf_filter=True,
logging_steps=1.0,
logging_strategy=steps,
lr_scheduler_kwargs={},
lr_scheduler_type=linear,
max_grad_norm=1.0,
max_steps=-1,
metric_for_best_model=None,
mp_parameters=,
neftune_noise_alpha=None,
no_cuda=False,
num_train_epochs=3.0,
optim=adamw_torch,
optim_args=None,
optim_target_modules=None,
output_dir=co_condenser_pretrain,
overwrite_output_dir=False,
past_index=-1,
per_device_eval_batch_size=8,
per_device_train_batch_size=16,
prediction_loss_only=False,
push_to_hub=False,
push_to_hub_model_id=None,
push_to_hub_organization=None,
push_to_hub_token=<PUSH_TO_HUB_TOKEN>,
ray_scope=last,
remove_unused_columns=False,
report_to=['wandb'],
resume_from_checkpoint=None,
run_name=co_condenser_pretrain,
save_on_each_node=False,
save_only_model=False,
save_safetensors=True,
save_steps=20000,
save_strategy=steps,
save_total_limit=None,
seed=42,
skip_memory_metrics=True,
split_batches=None,
tf32=None,
torch_compile=False,
torch_compile_backend=None,
torch_compile_mode=None,
torchdynamo=None,
tpu_metrics_debug=False,
tpu_num_cores=None,
use_cpu=False,
use_ipex=False,
use_legacy_prediction_loop=False,
use_mps_device=False,
warmup_ratio=0.1,
warmup_steps=0,
weight_decay=0.0,
)
[INFO|configuration_utils.py:724] 2024-03-30 22:20:55,489 >> loading configuration file /home/ubuntu/ejpark/checkpoint-260000/config.json
[INFO|configuration_utils.py:789] 2024-03-30 22:20:55,491 >> Model config BertConfig {
  "_name_or_path": "/home/ubuntu/ejpark/checkpoint-260000/",
  "architectures": [
    "BertForMaskedLM"
  ],
  "attention_probs_dropout_prob": 0.1,
  "classifier_dropout": null,
  "gradient_checkpointing": false,
  "hidden_act": "gelu",
  "hidden_dropout_prob": 0.1,
  "hidden_size": 768,
  "initializer_range": 0.02,
  "intermediate_size": 3072,
  "layer_norm_eps": 1e-12,
  "max_position_embeddings": 512,
  "model_type": "bert",
  "num_attention_heads": 12,
  "num_hidden_layers": 12,
  "pad_token_id": 0,
  "position_embedding_type": "absolute",
  "torch_dtype": "float32",
  "transformers_version": "4.39.1",
  "type_vocab_size": 2,
  "use_cache": true,
  "vocab_size": 61790
}

[INFO|tokenization_utils_base.py:2082] 2024-03-30 22:20:55,492 >> loading file vocab.txt
[INFO|tokenization_utils_base.py:2082] 2024-03-30 22:20:55,492 >> loading file added_tokens.json
[INFO|tokenization_utils_base.py:2082] 2024-03-30 22:20:55,492 >> loading file special_tokens_map.json
[INFO|tokenization_utils_base.py:2082] 2024-03-30 22:20:55,492 >> loading file tokenizer_config.json
[INFO|tokenization_utils_base.py:2082] 2024-03-30 22:20:55,492 >> loading file tokenizer.json
[INFO|modeling_utils.py:3280] 2024-03-30 22:20:55,571 >> loading weights file /home/ubuntu/ejpark/checkpoint-260000/model.safetensors
[INFO|configuration_utils.py:928] 2024-03-30 22:20:55,581 >> Generate config GenerationConfig {
  "pad_token_id": 0
}

[INFO|modeling_utils.py:4024] 2024-03-30 22:20:56,202 >> All model checkpoint weights were used when initializing BertForMaskedLM.

[INFO|modeling_utils.py:4032] 2024-03-30 22:20:56,202 >> All the weights of BertForMaskedLM were initialized from the model checkpoint at /home/ubuntu/ejpark/checkpoint-260000/.
If your task is similar to the task the model of the checkpoint was trained on, you can already use BertForMaskedLM for predictions without further training.
[INFO|configuration_utils.py:881] 2024-03-30 22:20:56,204 >> loading configuration file /home/ubuntu/ejpark/checkpoint-260000/generation_config.json
[INFO|configuration_utils.py:928] 2024-03-30 22:20:56,205 >> Generate config GenerationConfig {
  "pad_token_id": 0
}

03/30/2024 22:20:56 - INFO - modeling -   loading extra weights from local files
-------------------------------------------
loading json . . .
03/30/2024 22:21:17 - WARNING - __main__ -   Process rank: 1, device: cuda:1, n_gpu: 1distributed training: True, 16-bits training: False
-------------------------------------------
loading json . . .
/home/ubuntu/ejpark/new_index_diet__.json load complete! 484.8482084274292 seconds took.
03/30/2024 22:29:01 - INFO - trainer -   Initializing Gradient Cache Trainer
/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/accelerate/accelerator.py:432: FutureWarning: Passing the following arguments to `Accelerator` is deprecated and will be removed in version 1.0 of Accelerate: dict_keys(['dispatch_batches', 'split_batches', 'even_batches', 'use_seedable_sampler']). Please pass an `accelerate.DataLoaderConfiguration` instead: 
dataloader_config = DataLoaderConfiguration(dispatch_batches=None, split_batches=False, even_batches=True, use_seedable_sampler=True)
  warnings.warn(
/home/ubuntu/ejpark/new_index_diet__.json load complete! 486.0928454399109 seconds took.
/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/accelerate/accelerator.py:432: FutureWarning: Passing the following arguments to `Accelerator` is deprecated and will be removed in version 1.0 of Accelerate: dict_keys(['dispatch_batches', 'split_batches', 'even_batches', 'use_seedable_sampler']). Please pass an `accelerate.DataLoaderConfiguration` instead: 
dataloader_config = DataLoaderConfiguration(dispatch_batches=None, split_batches=False, even_batches=True, use_seedable_sampler=True)
  warnings.warn(
[INFO|trainer.py:1969] 2024-03-30 22:31:05,175 >> ***** Running training *****
[INFO|trainer.py:1970] 2024-03-30 22:31:05,176 >>   Num examples = 7,204
[INFO|trainer.py:1971] 2024-03-30 22:31:05,176 >>   Num Epochs = 3
[INFO|trainer.py:1972] 2024-03-30 22:31:05,176 >>   Instantaneous batch size per device = 16
[INFO|trainer.py:1975] 2024-03-30 22:31:05,176 >>   Total train batch size (w. parallel, distributed & accumulation) = 32
[INFO|trainer.py:1976] 2024-03-30 22:31:05,176 >>   Gradient Accumulation steps = 1
[INFO|trainer.py:1977] 2024-03-30 22:31:05,176 >>   Total optimization steps = 678
[INFO|trainer.py:1978] 2024-03-30 22:31:05,177 >>   Number of trainable parameters = 147,735,134
[INFO|integration_utils.py:723] 2024-03-30 22:31:05,178 >> Automatic Weights & Biases logging enabled, to disable set os.environ["WANDB_DISABLED"] = "true"
  0%|          | 0/678 [00:00<?, ?it/s]Traceback (most recent call last):
  File "/home/ubuntu/ejpark/Condenser/run_co_pre_training.py", line 221, in <module>
    main()
  File "/home/ubuntu/ejpark/Condenser/run_co_pre_training.py", line 211, in main
    trainer.train()
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/transformers/trainer.py", line 1780, in train
    return inner_training_loop(
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/transformers/trainer.py", line 2085, in _inner_training_loop
    for step, inputs in enumerate(epoch_iterator):
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/accelerate/data_loader.py", line 452, in __iter__
    current_batch = next(dataloader_iter)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 630, in __next__
    data = self._next_data()
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 674, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 54, in fetch
    return self.collate_fn(data)
  File "/home/ubuntu/ejpark/Condenser/data.py", line 173, in __call__
    return super(CoCondenserCollator, self).__call__(examples)
  File "/home/ubuntu/ejpark/Condenser/data.py", line 131, in __call__
    e_trunc = self._truncate(e['text'])
  File "/home/ubuntu/ejpark/Condenser/data.py", line 103, in _truncate
    tgt_len = self.max_seq_length - self.tokenizer.num_special_tokens_to_add(False)
TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'
Traceback (most recent call last):
  File "/home/ubuntu/ejpark/Condenser/run_co_pre_training.py", line 221, in <module>
    main()
  File "/home/ubuntu/ejpark/Condenser/run_co_pre_training.py", line 211, in main
    trainer.train()
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/transformers/trainer.py", line 1780, in train
    return inner_training_loop(
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/transformers/trainer.py", line 2085, in _inner_training_loop
    for step, inputs in enumerate(epoch_iterator):
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/accelerate/data_loader.py", line 452, in __iter__
    current_batch = next(dataloader_iter)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 630, in __next__
    data = self._next_data()
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 674, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 54, in fetch
    return self.collate_fn(data)
  File "/home/ubuntu/ejpark/Condenser/data.py", line 173, in __call__
    return super(CoCondenserCollator, self).__call__(examples)
  File "/home/ubuntu/ejpark/Condenser/data.py", line 131, in __call__
    e_trunc = self._truncate(e['text'])
  File "/home/ubuntu/ejpark/Condenser/data.py", line 103, in _truncate
    tgt_len = self.max_seq_length - self.tokenizer.num_special_tokens_to_add(False)
TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'
wandb: - 0.007 MB of 0.007 MB uploadedwandb: \ 0.007 MB of 0.007 MB uploadedwandb: | 0.007 MB of 0.007 MB uploadedwandb: / 0.007 MB of 0.007 MB uploadedwandb: - 0.007 MB of 0.021 MB uploadedwandb: \ 0.007 MB of 0.024 MB uploadedwandb: | 0.024 MB of 0.024 MB uploadedwandb: üöÄ View run effortless-dream-46 at: https://wandb.ai/3jp4rk/Condenser/runs/w6obiu37/workspace
wandb: Synced 5 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
wandb: Find logs at: ./wandb/run-20240330_222012-w6obiu37/logs
Exception in thread IntMsgThr:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 980, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.9/threading.py", line 917, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/wandb_run.py", line 300, in check_internal_messages
    self._loop_check_status(
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/wandb_run.py", line 224, in _loop_check_status
    local_handle = request()
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/interface/interface.py", line 844, in deliver_internal_messages
    return self._deliver_internal_messages(internal_message)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/interface/interface_shared.py", line 516, in _deliver_internal_messages
    return self._deliver_record(record)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/interface/interface_shared.py", line 459, in _deliver_record
    handle = mailbox._deliver_record(record, interface=self)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/lib/mailbox.py", line 455, in _deliver_record
    interface._publish(record)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/interface/interface_sock.py", line 51, in _publish
    self._sock_client.send_record_publish(record)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/lib/sock_client.py", line 221, in send_record_publish
    self.send_server_request(server_req)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/lib/sock_client.py", line 155, in send_server_request
    self._send_message(msg)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/lib/sock_client.py", line 152, in _send_message
    self._sendall_with_error_handle(header + data)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/wandb/sdk/lib/sock_client.py", line 130, in _sendall_with_error_handle
    sent = self._sock.send(data)
BrokenPipeError: [Errno 32] Broken pipe
wandb: - 0.007 MB of 0.007 MB uploadedwandb: \ 0.007 MB of 0.007 MB uploadedwandb: | 0.007 MB of 0.007 MB uploadedwandb: / 0.007 MB of 0.012 MB uploadedwandb: - 0.007 MB of 0.012 MB uploadedwandb: \ 0.012 MB of 0.012 MB uploadedwandb: üöÄ View run woven-mountain-45 at: https://wandb.ai/3jp4rk/Condenser/runs/vky4ar5t/workspace
wandb: Synced 5 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
wandb: Find logs at: ./wandb/run-20240330_222012-vky4ar5t/logs
[2024-03-30 22:40:59,392] torch.distributed.elastic.multiprocessing.api: [WARNING] Sending process 3270791 closing signal SIGTERM
[2024-03-30 22:41:09,470] torch.distributed.elastic.multiprocessing.api: [ERROR] failed (exitcode: 1) local_rank: 0 (pid: 3270790) of binary: /home/ubuntu/ejpark/con_venv/bin/python3.9
Traceback (most recent call last):
  File "/home/ubuntu/ejpark/con_venv/bin/torchrun", line 8, in <module>
    sys.exit(main())
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 346, in wrapper
    return f(*args, **kwargs)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/distributed/run.py", line 806, in main
    run(args)
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/distributed/run.py", line 797, in run
    elastic_launch(
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/distributed/launcher/api.py", line 134, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/home/ubuntu/ejpark/con_venv/lib/python3.9/site-packages/torch/distributed/launcher/api.py", line 264, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
run_co_pre_training.py FAILED
------------------------------------------------------------
Failures:
  <NO_OTHER_FAILURES>
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2024-03-30_22:40:59
  host      : gpu-1.novalocal
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 3270790)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
./train_co.sh: 15: --learning_rate: not found
